src/core.c/CompUnit/Repository/Perl5.pm6

```diff
@@ -1,51 +1,56 @@
 class CompUnit::Repository::Perl5 does CompUnit::Repository {
-    method need(
-        CompUnit::DependencySpecification $spec,
-        CompUnit::PrecompilationRepository $precomp = self.precomp-repository(),
-        --> CompUnit:D)
-    {
+    my $perl5-dependency :=
+      CompUnit::DependencySpecification.new(:short-name<Inline::Perl5>);
+
+    method need(CompUnit::Repository::Perl5:D:
+      CompUnit::DependencySpecification $spec,
+      CompUnit::PrecompilationRepository $precomp,
+    --> CompUnit:D) {
         if $spec.from eq 'Perl5' {
-            my $compunit = $*REPO.need(CompUnit::DependencySpecification.new(:short-name<Inline::Perl5>));
-            my $perl5 := $compunit.handle.globalish-package<Inline>.WHO<Perl5>.default_perl5;
+            CATCH {
+                when X::CompUnit::UnsatisfiedDependency {
+                    X::NYI::Available.new(
+                      :available('Inline::Perl5'),
+                      :feature('Perl 5')
+                    ).throw;
+                }
+            }
+
+            my $compunit := $*REPO.need($perl5-dependency);
+            my $perl5    := $compunit.handle.globalish-package<Inline>.WHO<Perl5>.default_perl5;
 
             if $*RAKUDO_MODULE_DEBUG -> $RMD {
-                $RMD("Loading {$spec.short-name} via Inline::Perl5");
+                $RMD("Loading $spec.short-name() via Inline::Perl5");
             }
+            my $short-name := $spec.short-name;
             my $handle := $perl5.require(
-                $spec.short-name,
-                $spec.version-matcher !== True ?? $spec.version-matcher.Num !! Num,
-                :handle
-            );
-            return CompUnit.new(
-                :short-name($spec.short-name),
-                :$handle,
-                :repo(self),
-                :repo-id($spec.short-name),
-                :from($spec.from),
+              $short-name,
+              $spec.version-matcher !== True
+                ?? $spec.version-matcher.Numi
```

Comment:
`.Numi`? Are we sure this code runs and works?

Reply:
Oops.  Fixed with https://github.com/rakudo/rakudo/commit/be98708fd5  Note that the condition was wonky, so I fixed that as well.